prestart.js

```diff
@@ -1,43 +1,359 @@
+
+const fs = require('fs');
+
 ig.module('game.feature.gui.teleport')
-	.requires('dom.ready', 'impact.feature.gui.gui')
+	.requires('dom.ready', 'impact.feature.gui.gui','game.feature.model.game-model','game.feature.combat.combat')
 	.defines(() => {
+		
+		
+		// ----------------- Local Variables --------------------------------------------------------------------
+
+		let modInterface;
+		let interfaceToggle;
+		
+		let modState = 0;
+		const MOD_STATE = {
+			ALL_HIDDEN: 0, 
+			TOGGLER_SHOWN: 1,
+			MOD_SHOWN: 2
+		}
+
+		const markerValues = ['up', 'down', 'left', 'right', 'landmark'];
+		let mapValues = [];
+
+
+		// ---------------------- init map list Variable
+		dir = './assets/data/maps/';
+
+		/**
+		 * @description get all teleportable maps recursively
+		 * @param {string} d start directory
+		 */
+		function getJsonFileNamesRecursive(d){
+			fs.readdir(d, (err, files) => {
+				files.forEach(file => {
+					if( file.search('.json') > 0 ){
+						map = d + '/' + file.replace('.json','');
+						mapValues.push( map.replace(dir,'').substr(1) );
+					}
+					else{
+						getJsonFileNamesRecursive(d+'/'+file);
+					}
+				});
+			});
+		}
+
+		getJsonFileNamesRecursive(dir);
+		// ---------------- Local Functions --------------------------------------------------------------------
+
+		/**
+		 * @description checks if is a valid map before teleporting
+		 * @param {string} map Input string from textbox 
+		 * @param {string} marker Map position to start 
+		 */
+		function teleportIfExists(map,marker){
+			mapPath = map.toPath(ig.root + 'data/maps/', '.json');	//From 
```

Comment:
Note that `path.join` is not required here because `mapPath` is a URL, not an FS path.